src/content/index.ts

```diff
@@ -27,21 +27,24 @@ function embedScript() {
   script.remove()
 }
 
+function sendCurrentState() {
+  const locale = window.sessionStorage.getItem(KEY) || null
+  const message: MessageType = { type: 'sync', data: locale }
+  browser.runtime.sendMessage(message)
+}
+
 function handleMessage({ type, data }: MessageType) {
   switch (type) {
     case 'update':
-      if (data === window.localStorage.getItem(KEY)) break
-      if (data) window.localStorage.setItem(KEY, data)
-      else window.localStorage.removeItem(KEY)
+      if (data === window.sessionStorage.getItem(KEY)) break
+      if (data) window.sessionStorage.setItem(KEY, data)
+      else window.sessionStorage.removeItem(KEY)
       window.location.reload()
       break
-
-    case 'get':
-      const locale = window.localStorage.getItem(KEY) || null
-      browser.runtime.sendMessage({ type: 'get', data: locale })
-      break
   }
 }
 
 browser.runtime.onMessage.addListener((message) => handleMessage(message))
+
 embedScript()
+sendCurrentState()
```

Comment:
Why do we need to send a `sync` message on page load?

Reply:
It's not sync actually, the function is a non blocking promise